src/error.ts

```diff
@@ -714,7 +714,7 @@ export function isRetryableError(error: MongoError): boolean {
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     (typeof error.code === 'number' && RETRYABLE_ERROR_CODES.has(error.code!)) ||
     error instanceof MongoNetworkError ||
-    !!error.message.match(/not master/) ||
+    !!error.message.match(/not primary/) ||
```

Comment:
I would create a constant for the regex `(not primary|not master)` and use it here.